mesh/generator/generator.go

```diff
@@ -68,6 +67,16 @@ func BuildMeshMap(ctx context.Context, o mesh.Options, gi *mesh.AppenderGlobalIn
 		}
 	}
 
+	// get istio status components (istiod, grafana, prometheus, tracing)
+	istioStatus, err := gi.IstioStatusGetter.GetStatus(ctx)
+	mesh.CheckError(err)
+
+	// convert istio status slice into map
+	healthData := map[string]string{}
+	for _, data := range istioStatus {
+		healthData[data.Name+data.Namespace+data.Cluster] = data.Status
```

Comment:
Added a key for this. I made it a struct so that someone using it is less likely to pass in params in the wrong order since all the key params are strings.